#!/usr/bin/env python
PACKAGE = "obstacle_avoidance"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kiteres_iranya",    str_t,    0, "milyen iranyban keruljuk az akakdalyt jobbra vagy balra ", "balra")
gen.add("kiteres_hossza",    double_t,    0, "milyen hosszu legyen az elkerules elso szakasza [m] ", 8.0,  0.0, 50.0)
gen.add("oldaliranyu_eltolas", double_t, 0, "Az oldaliranyu elkerules nagysaga [m] ",    3.0, 0.0,  10.0 )
gen.add("elkerules_hossza",    double_t,    0, "milyen hosszu legyen az elkerules kozepso szakasza [m]",2.0,0.0,50.0)
gen.add("visszateres_hossza",    double_t,    0, "milyen hosszu legyen az elkerules utolso szakasza [m]",8.0,0.0,50.0)
gen.add("distance_delta",    double_t,    0, "az ujratervezett pontok tavolsaga [m]",0.8,0.0,3.0)
gen.add("lookahead", int_t,    0, "A vizsgalat a kocsi elott hany waypointon keresztul menjen vegbe [db]",40,0,100)
gen.add("polygon_size_threshold",    double_t,    0, "Az euklidezi klasztertol kapott poligon minimum kerulete [m]",0.5,0.0,5.0)
gen.add("presence_threshold",    int_t,    0, "Minimum hany elofordulast kerunk a waypointol hogy valid legyen [db]",5,0,100)


exit(gen.generate(PACKAGE, "obstacle_avoidance", "Params"))